File: 2026-03/src/day_09/boj_10799/Main.java
package day_09.boj_10799;

import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        Deque<Character> stack = new ArrayDeque<>();
        int cnt = 0;

        String s = br.readLine();
        char prv = '\u0000';

        for (char c : s.toCharArray()) {
            if (c == '(') {
                stack.push(c);
            } else {
                if (prv == '(') {
                    stack.pop();
                    cnt += stack.size();
                } else {
                    stack.pop();
                    cnt++;
                }
            }

            prv = c;
        }

        System.out.println(cnt);
    }
}

File: 2026-03/src/day_09/boj_10799/Main.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);

    stack<char> st;
    int cnt = 0;

    string s;
    cin >> s;
    char prv = '\0';

    for (char c : s) {
        if (c == '(') {
            st.push(c);
        } else {
            if (prv == '(') {
                st.pop();
                cnt += st.size();
            } else {
                st.pop();
                cnt++;
            }
        }

        prv = c;
    }

    cout << cnt;
}
